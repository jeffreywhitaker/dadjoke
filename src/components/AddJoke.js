// import dependencies
import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { connect } from "react-redux";
import styled from "styled-components";

import { addJoke } from "../actions/actions";

// add joke component
function AddJoke({ addJoke }) {
  const history = useHistory();

  // local state for adding new joke
  const [newJoke, setNewJoke] = useState({
    dadjokequestion: "",
    dadjokeanswer: "",
    isprivate: false
  });

  // call add joke function
  const callAddJoke = e => {
    console.log("stuff is happening");
    e.preventDefault();
    addJoke(newJoke);
    setNewJoke({ dadjokequestion: "", dadjokeanswer: "", isprivate: false });
    history.push("/jokes");
  };

  // handle change values, save to local state
  const handleValueChange = e => {
    setNewJoke({
      ...newJoke,
      [e.target.name]: e.target.value
    });
  };

  // render the following
  return (
    <AddJokeCardDiv>
      <input
        type="text"
        name="dadjokequestion"
        value={newJoke.dadjokequestion}
        onChange={handleValueChange}
      />
      <input
        type="text"
        name="dadjokeanswer"
        value={newJoke.dadjokeanswer}
        onChange={handleValueChange}
      />
      <button onClick={callAddJoke}>Add</button>
    </AddJokeCardDiv>
  );
}

// connect component to redux store
const mapStateToProps = state => {
  return {
    jokes: state.jokes.jokes
  };
};

// export component
export default connect(mapStateToProps, { addJoke })(AddJoke);

// styled components
const AddJokeCardDiv = styled.div`
  width: 80%
  margin: 20px auto
  background: cyan
  border-radius: 15px
  padding: 5px 15px
`;
