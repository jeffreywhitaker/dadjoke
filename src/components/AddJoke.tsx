// main imports
import React, { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { connect } from 'react-redux'
import styled from 'styled-components'

import { addJoke } from '../actions/actions'

// add joke component
// TODO: make work export const AddJoke: React.FC = (): ReactElement =>
export const AddJoke: React.FC = () => {
  const history = useHistory()

  const blankJoke = {
    dadjokequestion: '',
    dadjokeanswer: '',
    isprivate: false,
  }
  // local state for adding new joke
  const [newJoke, setNewJoke] = useState(blankJoke)

  // call add joke function
  const callAddJoke = (e: { preventDefault: () => void }) => {
    e.preventDefault()
    console.log('new joke:')
    console.log(newJoke)
    addJoke(newJoke)
    setNewJoke(blankJoke)
    history.push('/publicjokes')
  }

  // handle change values, save to local state
  const handleValueChange = (e: {
    target: { type: string; checked: any; value: any; name: any }
  }) => {
    const value =
      e.target.type === 'checkbox' ? e.target.checked : e.target.value
    setNewJoke({
      ...newJoke,
      [e.target.name]: value,
    })
    console.log(newJoke)
  }

  // render the following
  return (
    <AddJokeCardDiv>
      <span>Q: </span>
      <input
        type="text"
        name="dadjokequestion"
        value={newJoke.dadjokequestion}
        onChange={handleValueChange}
      />
      <br />
      <span>A: </span>
      <input
        type="text"
        name="dadjokeanswer"
        value={newJoke.dadjokeanswer}
        onChange={handleValueChange}
      />
      <br />
      <span>is Private?</span>
      <input
        type="checkbox"
        name="isprivate"
        checked={newJoke.isprivate}
        onChange={handleValueChange}
      />
      <button onClick={callAddJoke}>Add</button>
    </AddJokeCardDiv>
  )
}

// connect component to redux store
const mapStateToProps = (state: {
  jokeReducer: { jokes: Array<Record<string, unknown>> }
  loginReducer: { username: string }
}) => {
  return {
    jokes: state.jokeReducer.jokes,
    usernameFromState: state.loginReducer.username,
  }
}

// // props
// type PropsFromRedux = ConnectedProps<typeof connector>
// interface Props extends PropsFromRedux {
//   addJoke: any,
//   usernameFromState: string,
// }

// export component
const connector = connect(mapStateToProps, { addJoke })
export default connector(AddJoke)

// styled components
const AddJokeCardDiv = styled.div`
  width: 80%;
  margin: 20px auto;
  background: lightblue;
  border-radius: 15px;
  padding: 5px 15px;
`
