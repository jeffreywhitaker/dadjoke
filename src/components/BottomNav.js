// import dependencies
import React from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

// import functions

// NavBar component
function BottomNav() {
  return (
    <NavBarSection>
      {window.location.pathname === '/jokes/' && (
        <NavLinkStyled exact to={'/jokes/add'}>
          Add Public Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/jokes/add' && (
        <NavLinkStyled exact to={'/jokes'}>
          Don't Add New Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/profile/' && (
        <NavLinkStyled exact to={'/profile/add'}>
          Add Private Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/profile/add' && (
        <NavLinkStyled exact to={'/profile'}>
          Don't Add New Joke
        </NavLinkStyled>
      )}
    </NavBarSection>
  )
}

// export NavBar
export default BottomNav

// styled components
const NavBarSection = styled.section`
  display: flex
  justify-content: space-around
  padding: 10px 0
`

const NavLinkStyled = styled(NavLink)`
  background: lightpink;
  text-decoration: none;
  color: black;
  padding: 5px 10px;
  &.active {
    filter: brightness(75%);
  }
`
