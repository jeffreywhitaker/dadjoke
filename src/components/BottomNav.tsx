// import dependencies
import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

// import functions

// NavBar component
function BottomNav(props: { isLoggedIn: boolean, location: { pathname: string} }) {
  // destructure props
  const { isLoggedIn } = props

  // logged out
  if (props.location.pathname === '/privatejokes' && !isLoggedIn) {
    return (
      <NavBarSection>
        <span>You must be logged in to add a private joke!</span>
      </NavBarSection>
    )
  }

  return (
    <NavBarSection>
      {window.location.pathname === '/publicjokes' && (
        <NavLinkStyled exact to={'/publicjokes/add'}>
          Add Public Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/publicjokes/add' && (
        <NavLinkStyled exact to={'/publicjokes'}>
          Don't Add New Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/privatejokes' && (
        <NavLinkStyled exact to={'/privatejokes/add'}>
          Add Private Joke
        </NavLinkStyled>
      )}
      {window.location.pathname === '/privatejokes/add' && (
        <NavLinkStyled exact to={'/privatejokes'}>
          Don't Add New Joke
        </NavLinkStyled>
      )}
    </NavBarSection>
  )
}

// connect component to redux store
const mapStateToProps = (state: { loginReducer: { isLoggedIn: boolean } }) => {
  return {
    isLoggedIn: state.loginReducer.isLoggedIn,
  }
}

// export NavBar
export default connect(mapStateToProps, {})(BottomNav)

// styled components
const NavBarSection = styled.section`
  display: flex;
  justify-content: space-around;
  padding: 10px 0;
`

const NavLinkStyled = styled(NavLink)`
  background: lightpink;
  text-decoration: none;
  color: black;
  padding: 5px 10px;
  &.active {
    filter: brightness(75%);
  }
  border-radius: 5px;
`
