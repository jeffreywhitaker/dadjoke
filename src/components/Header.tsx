// import dependencies
import React, { useEffect, useState } from 'react'
import { connect, ConnectedProps } from 'react-redux'
import { NavLink, useHistory } from 'react-router-dom'
import styled from 'styled-components'

// bootstrap
import Button from 'react-bootstrap/Button'
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'
import Modal from 'react-bootstrap/Modal'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav'
import ToggleButton from 'react-bootstrap/ToggleButton'
import ButtonGroup from 'react-bootstrap/ButtonGroup'
import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
import Tooltip from 'react-bootstrap/Tooltip'

// import functions
import { Joke } from '../types/types'
import jokesData from '../ajax/jokesData'
import { userLogout } from '../actions/actions'

// Header component
export const Header: React.FC<Props> = (props: Props) => {
  // destructure props
  const { isLoggedIn, userLogout, username } = props

  // modal
  // local state for adding new joke
  function determineIfPrivate(): boolean {
    if (window.location.pathname === '/publicjokes') {
      return false
    } else return true
  }
  const blankJoke: Joke = {
    dadjokequestion: '',
    dadjokeanswer: '',
    keywords: [],
    isprivate: determineIfPrivate(),
  }
  const [showAddJokeModal, setShowAddJokeModal] = useState(false)
  const [newJoke, setNewJoke] = useState(blankJoke)

  const handleClose = () => setShowAddJokeModal(false)
  const handleSetIsPrivate = (val: boolean) =>
    setNewJoke({ ...newJoke, isprivate: val })

  // call add joke function
  const callAddJoke = (e: { preventDefault: () => unknown }) => {
    e.preventDefault()
    setShowAddJokeModal(false)
    console.log('new joke:')
    console.log(newJoke)
    // addJoke(newJoke)
    jokesData
      .addNewJoke(newJoke)
      .then(() => {
        setNewJoke(blankJoke)
        // RELOAD THE APP RATHER THAN BOTHER PIPING NEW JOKE TO THE CORRECT COMPONENT
        window.location.reload()
      })
      .catch((err) => {
        window.alert('Unable to add new joke: ' + err)
      })
  }

  useEffect(() => {
    console.log('isprivate', newJoke.isprivate)
  }, [newJoke])

  // handle change values, save to local state
  const handleValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value =
      e.target.type === 'checkbox' ? e.target.checked : e.target.value
    setNewJoke({
      ...newJoke,
      [e.target.name]: value,
    })
    console.log(newJoke)
  }

  // handle changing the keywords
  const handleKeywordsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const stringOfKeywords = e.currentTarget.value
    setNewJoke({
      ...newJoke,
      keywords: stringOfKeywords
        .trim()
        .replace(/\s/g, '')
        .split(','),
    })
  }

  const handleShow = () => setShowAddJokeModal(true)

  // use history
  const history = useHistory()

  console.log('is logged in', isLoggedIn)

  useEffect(() => {
    console.log('new joke changed: ', newJoke)
  }, [newJoke])

  // helper functions
  const handleLogout = (e: { preventDefault: () => void }) => {
    console.log('handle logout called')
    e.preventDefault()
    if (window.confirm('Are you sure you want to log out?')) {
      userLogout()
      history.push('/publicjokes')
    }
  }

  // return Header
  return (
    <HeaderWrapper>
      <Navbar collapseOnSelect expand="md" bg="dark" variant="dark">
        <Navbar.Brand as={NavLink} to="/publicjokes">
          JeffsDadJokes
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto">
            <Nav.Link as={NavLink} to="/publicjokes">
              Public Jokes
            </Nav.Link>
            <Nav.Link as={NavLink} to="/privatejokes">
              Private Jokes
            </Nav.Link>
            {isLoggedIn ? (
              <Nav.Link as={NavLink} to={`/profile/${username}`}>
                {username}'s Profile
              </Nav.Link>
            ) : null}
          </Nav>
          <Nav>
            {isLoggedIn ? (
              <>
                <Button
                  variant="primary"
                  className="button"
                  onClick={handleShow}
                >
                  Add Joke
                </Button>
                &nbsp;
                <Button
                  variant="danger"
                  className="button"
                  onClick={handleLogout}
                >
                  Log Out
                </Button>
              </>
            ) : (
              <>
                <Nav.Link as={NavLink} to="/login">
                  Login
                </Nav.Link>
                <Nav.Link as={NavLink} to="/signup">
                  Signup
                </Nav.Link>
              </>
            )}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
      <Modal show={showAddJokeModal} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Add Joke</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text>Q</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              aria-label="Add DadJoke Question"
              name="dadjokequestion"
              value={newJoke.dadjokequestion}
              onChange={handleValueChange}
            />
          </InputGroup>
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text>A</InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              aria-label="Add DadJoke Answer"
              name="dadjokeanswer"
              value={newJoke.dadjokeanswer}
              onChange={handleValueChange}
            />
          </InputGroup>
          <InputGroup className="mb-3">
            <InputGroup.Prepend>
              <InputGroup.Text>
                <span>Keywords</span> &nbsp;
                <OverlayTrigger
                  key={`keywords`}
                  placement="top"
                  overlay={
                    <Tooltip id={`keywords`}>
                      Keywords should be a comma-seperated list of words
                    </Tooltip>
                  }
                >
                  <i className="fas fa-info-circle"></i>
                </OverlayTrigger>
              </InputGroup.Text>
            </InputGroup.Prepend>
            <FormControl
              aria-label="Add DadJoke Answer"
              name="dadjokeanswer"
              onChange={handleKeywordsChange}
            />
          </InputGroup>

          <ButtonGroup toggle>
            <ToggleButton
              type="radio"
              name="radio"
              checked={!newJoke.isprivate}
              value={false}
              onClick={() => handleSetIsPrivate(false)}
            >
              Public
            </ToggleButton>
            <ToggleButton
              type="radio"
              checked={newJoke.isprivate}
              value={true}
              onClick={() => handleSetIsPrivate(true)}
            >
              Private
            </ToggleButton>
          </ButtonGroup>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="primary" onClick={callAddJoke}>
            Add Joke
          </Button>
        </Modal.Footer>
      </Modal>
    </HeaderWrapper>
  )
}

// connect component to redux store
const mapStateToProps = (state: {
  loginReducer: { isLoggedIn: boolean; username: string }
}) => {
  return {
    isLoggedIn: state.loginReducer.isLoggedIn,
    username: state.loginReducer.username,
  }
}

// export component
const connector = connect(mapStateToProps, { userLogout })
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux
export default connector(Header)

const HeaderWrapper = styled.section`
  .button {
    max-width: 150px;
  }
`
