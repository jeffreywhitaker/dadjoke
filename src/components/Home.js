// import dependencies
import React from "react";

// Home component
function Home() {
  return <p>This is the home component.</p>;
}

// export Home
export default Home;
