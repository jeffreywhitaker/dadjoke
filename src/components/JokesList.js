// import dependencies
import React, { useEffect } from "react";
import { connect } from "react-redux";

// import actions
import { getPublicJokes } from "../actions/actions";
import SingleJokeCard from "./SingleJokeCard";

// login page component
function JokesList({ getPublicJokes, jokes }) {
  console.log("jokes", jokes);
  // get jokes on page load
  useEffect(() => {
    getPublicJokes();
  }, [getPublicJokes]);

  // render the following
  return (
    <>
      {jokes ? (
        jokes.map(joke => {
          return <SingleJokeCard joke={joke} key={joke.dadjokeid} />;
        })
      ) : (
        <p>No jokes are in database - add them now!</p>
      )}
    </>
  );
}

// connect component to redux store
const mapStateToProps = state => {
  return {
    jokes: state.jokes.jokes
  };
};

// export component
export default connect(mapStateToProps, { getPublicJokes })(JokesList);
