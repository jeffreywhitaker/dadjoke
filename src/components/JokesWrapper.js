// import dependencies
import React, { useEffect, useState } from 'react'
import { connect } from 'react-redux'
import { useLocation } from 'react-router-dom'
import '../styles/loadingSpinner.css'
import styled from 'styled-components'
import cloneDeep from 'clone-deep'

// bootstrap
import Button from 'react-bootstrap/Button'

// import actions
import jokesData from '../ajax/jokesData'
import SingleJokeCard from './SingleJokeCard'
import Loading from './Loading'
import Pagination from './small/Pagination'

// joke display page component
function JokesWrapper({ isLoggedIn, username }) {
  const location = useLocation()
  console.log('location is: ', location)
  // set up state
  const [isLoading, setIsLoading] = useState(true)
  const [jokes, setJokes] = useState([])
  const [searchString, setSearchString] = useState('')
  const [display, setDisplay] = useState({
    heading: '',
  })

  const [criteria, setCriteria] = useState({
    sortBy: '-createdAt',
    resultsPerPage: '5',
    searchString: '',
    page: 1,
  })
  const [hasNextPage, setHasNextPage] = useState(false)

  // get jokes on page load
  useEffect(() => {
    console.log('calling useEffect with criteria', criteria)
    let publicOrPrivate

    if (location.pathname === '/publicjokes') {
      publicOrPrivate = 'public'
      setDisplay({ ...display, heading: 'Public Jokes' })
    } else {
      publicOrPrivate = 'private'
      setDisplay({ ...display, heading: `${username}'s Jokes` })
    }

    if (location.pathname === '/publicjokes' || isLoggedIn) {
      jokesData
        .getJokes(criteria, publicOrPrivate)
        .then((res) => {
          console.log('jokes response:', res)
          setJokes(res.data.jokes)
          setHasNextPage(res.data.hasNextPage)
          setIsLoading(false)
        })
        .catch((err) => {
          window.alert('Unable to find jokes: ', err)
        })
    }
  }, [criteria])

  // set keyword search by search string, but with debounce
  const setKeywordSearch = () => {
    setCriteria({ ...criteria, searchString: searchString })
  }

  const updateJokeKarma = (jokeID, newKarma, newVote) => {
    console.log('updateJokeKarma: ', jokeID, newKarma)
    const updatedJokes = cloneDeep(jokes)
    updatedJokes.forEach((joke) => {
      if (joke._id === jokeID) {
        joke.karma = newKarma
        joke.userVote = newVote.toString()
      }
    }),
      setJokes(updatedJokes)
  }

  const updateFollowJokeCreator = (jokeCreator, isFollowing) => {
    console.log('updateFollowJokeCreator', jokeCreator, isFollowing)
    const updatedJokes = cloneDeep(jokes)
    updatedJokes.forEach((joke) => {
      if (joke.username === jokeCreator) {
        joke.userFollowingCreator = isFollowing
      }
    })
    setJokes(updatedJokes)
  }

  const handleSortByChange = (e) => {
    console.log('sort by changed to: ', e.currentTarget.value)
    setCriteria({
      ...criteria,
      sortBy: e.currentTarget.value,
    })
  }

  const handleResultsPerPageChange = (e) => {
    setCriteria({
      ...criteria,
      page: 1,
      resultsPerPage: e.currentTarget.value,
    })
  }

  const handlePageDown = () => {
    setCriteria({
      ...criteria,
      page: criteria.page - 1,
    })
  }

  const handlePageUp = () => {
    setCriteria({
      ...criteria,
      page: criteria.page + 1,
    })
  }

  const updateJokeDetails = (jokeID, res) => {
    // replace values of updated joke with response
    let index
    for (let i = 0; i < jokes.length; i++) {
      if (jokes[i]._id == jokeID) {
        index = i
        break
      }
    }
    const updatedJokes = cloneDeep(jokes)
    updatedJokes[index] = res.data
    setJokes(updatedJokes)
    //old
    // const updatedJokes = cloneDeep(jokes).filter((joke) => joke._id !== jokeID)
    // updatedJokes.push(res.data)
    // setJokes(updatedJokes)
  }

  // login check
  if (location.pathname === '/privatejokes' && !isLoggedIn) {
    return (
      <p style={{ textAlign: 'center' }}>
        You must be logged in to see your private jokes. Please log in above!
      </p>
    )
  }

  // render the following if checks pass
  return (
    <JokeWrapper>
      <h1 className="title">{display.heading}</h1>
      {isLoading ? (
        <Loading />
      ) : (
        <div>
          <SortDiv>
            <div className="sortOptions">
              Sort by:{' '}
              <select
                name="sortBy"
                id="sortBy"
                value={criteria.sortBy}
                onChange={handleSortByChange}
              >
                <option value="-createdAt">Newest to Oldest</option>
                <option value="createdAt">Oldest to Newest</option>
                <option value="-karma">Karma, Highest to Lowest</option>
                <option value="karma">Karma, Lowest to Highest</option>
              </select>
              &nbsp;&nbsp; Results:{' '}
              <select
                name="resultsPerPage"
                id="resultsPerPage"
                value={criteria.resultsPerPage}
                onChange={handleResultsPerPageChange}
              >
                <option value="2">2</option>
                <option value="5">5</option>
                <option value="10">10</option>
                <option value="20">20</option>
              </select>
              &nbsp;&nbsp; Keyword:{' '}
              <input
                name="search"
                value={searchString}
                onChange={(e) => setSearchString(e.currentTarget.value)}
              />
              &nbsp;&nbsp;
              <button onClick={setKeywordSearch}>Set Keyword</button>
            </div>

            <Pagination
              criteria={criteria}
              hasNextPage={hasNextPage}
              handlePageDown={handlePageDown}
              handlePageUp={handlePageUp}
            />
          </SortDiv>
          {jokes.length < 1 ? (
            <div className="emptyWrapper">
              <img src="/img/maninrain.png" className="image" />
              <h2>No jokes are in the database - add them now!</h2>
            </div>
          ) : (
            jokes.map((joke) => {
              return (
                <SingleJokeCard
                  joke={joke}
                  username={username}
                  key={joke.dadjokequestion}
                  updateJokeKarma={updateJokeKarma}
                  updateJokeDetails={updateJokeDetails}
                  updateFollowJokeCreator={updateFollowJokeCreator}
                />
              )
            })
          )}
        </div>
      )}
    </JokeWrapper>
  )
}

// connect component to redux store
const mapStateToProps = (state) => {
  return {
    username: state.loginReducer.username,
    jokesUpvoted: state.loginReducer.jokesUpvoted,
    jokesDownvoted: state.loginReducer.jokesDownvoted,
    isLoggedIn: state.loginReducer.isLoggedIn,
  }
}

// export component
export default connect(mapStateToProps, {})(JokesWrapper)

// styled components
const JokeWrapper = styled.article`
  display: flex;
  flex-direction: column;
  padding: 10px;

  > .title {
    text-align: center;
    font-size: 20px;
    background: lightblue;
    width: 50%;
    margin: 0 auto;
    border-radius: 15px;
    padding: 10px 0;
  }

  .emptyWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;

    .image {
      max-width: 400px;
    }
  }
`

const SortDiv = styled.div`
  background: lightblue;
  margin: 0 auto;
  border-radius: 6px;
  display: flex;
  padding: 10px;
  margin-top: 10px;
  justify-content: space-between;

  > .sortOptions {
    padding-left: 20px;
  }
`
