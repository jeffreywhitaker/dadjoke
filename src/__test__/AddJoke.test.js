import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

import { AddJoke } from '../components/AddJoke'
// import 'jest-styled-components';

describe('AddJoke component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <AddJoke addJoke={() => {}} />
        </Router>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
