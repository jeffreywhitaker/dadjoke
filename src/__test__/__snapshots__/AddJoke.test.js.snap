// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddJoke component renders correctly 1`] = `
<div
  className="sc-bdVaJa hkJhzp"
>
  <span>
    Q: 
  </span>
  <input
    name="dadjokequestion"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    A: 
  </span>
  <input
    name="dadjokeanswer"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <span>
    is Private?
  </span>
  <input
    checked={false}
    name="isprivate"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
